pkg/api/utils.go

```diff
@@ -28,32 +29,49 @@ import (
 var sanitizer = strings.NewReplacer(`&`, "&amp;", `<`, "&lt;", `>`, "&gt;")
 
 func HandleInternalError(response *restful.Response, req *restful.Request, err error) {
-	_, fn, line, _ := runtime.Caller(1)
-	klog.Errorf("%s:%d %v", fn, line, err)
-	http.Error(response, sanitizer.Replace(err.Error()), http.StatusInternalServerError)
+	handle(http.StatusInternalServerError, response, err)
 }
 
 // HandleBadRequest writes http.StatusBadRequest and log error
 func HandleBadRequest(response *restful.Response, req *restful.Request, err error) {
-	_, fn, line, _ := runtime.Caller(1)
-	klog.Errorf("%s:%d %v", fn, line, err)
-	http.Error(response, sanitizer.Replace(err.Error()), http.StatusBadRequest)
+	handle(http.StatusBadRequest, response, err)
 }
 
 func HandleNotFound(response *restful.Response, req *restful.Request, err error) {
-	_, fn, line, _ := runtime.Caller(1)
-	klog.Errorf("%s:%d %v", fn, line, err)
-	http.Error(response, sanitizer.Replace(err.Error()), http.StatusNotFound)
+	handle(http.StatusNotFound, response, err)
 }
 
 func HandleForbidden(response *restful.Response, req *restful.Request, err error) {
-	_, fn, line, _ := runtime.Caller(1)
-	klog.Errorf("%s:%d %v", fn, line, err)
-	http.Error(response, sanitizer.Replace(err.Error()), http.StatusForbidden)
+	handle(http.StatusForbidden, response, err)
+}
+
+func HandleUnauthorized(response *restful.Response, err error) {
```

Comment:
Why there is no `request *restful.Request` in function signature? request info may needed in the future.

Reply:
refined